### PR TITLE
Add --output and --in-place options

### LINKED ISSUE #886: in-place transformation for refmt
Some switch that allows the transformation of a file in-place, since `cat foo.re | refmt ... > foo.re` doesn't work because of how unix pipes work.

### PR DESCRIPTION
**Pull request by tekknolagi**
`--output`/`-o` allow the user to specify an output file for the results of refmt.
`--in-place` sets the output the file to the input file. Fixes #886

This is currently a pretty gross solution. Looking into fixing that now.

### CODE REVIEW

**Comment by tekknolagi** [comment]
@yunxing I think this is ready.

File: src/printer_maker.ml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        val makePrinter : string option ->
+                          string ->
+                          bool ->
+                          out_channel ->
+                          Format.formatter ->
+                          ((t * Reason_pprint_ast.commentWithCategory) -> unit)
+    end
+
+let prepare_output_file output_file =
```
Comment (yunxing) [inline src/printer_maker.ml]: let prepare_output_file = function 
| Some name -> open_out name
| None -> ()

Comment (tekknolagi) [inline src/printer_maker.ml]: good point.

File: src/refmt_impl.ml [hunk 2]
```diff
@@ -138,118 +114,48 @@ let () =
...
-        | Some "none" -> (fun (ast, comments) -> ())
-        | Some "ml" -> Reason_toolchain.ML.print_canonical_implementation_with_comments
-        | Some "re" -> Reason_toolchain.JS.print_canonical_implementation_with_comments
-        | Some s -> (
-          raise (Invalid_config ("Invalid --print setting for implementation '" ^ s ^ "'."))
-        )
-      in
-      thePrinter (ast, comments)
+    let (ast, parsedAsML) = Printer.parse !prse use_stdin filename in
+    (* As it turns out, calling formatter_of_out_channel on stdout results in
```
Comment (yunxing) [inline src/refmt_impl.ml]: Is this still true? I see later on you are calling formatter_of_out_channel on output_chan and it could be stdout.

Comment (tekknolagi) [inline src/refmt_impl.ml]: Ah, whoops. Did not mean to leave this comment in. I think I mislabeled the true culprit.

**Comment by yunxing** [comment]
Looks good. Could you also help fix the conflicts?

**Comment by tekknolagi** [comment]
Sure. I can try rebasing master locally

**Comment by tekknolagi** [comment]
Fixed!

**Comment by yunxing** [comment]
Thanks! Let's wait for CI and we can push.

**Comment by tekknolagi** [comment]
How has CI not started 8 hours later?

**Comment by yunxing** [comment]
@tekknolagi It has been an issue for us -- all projects in FB share the same worker pool.

**Pull request merged**